### DISCUSSION TITLE
Thanks for your work

### DISCUSSION THREAD

**Discussion by waynehamadi** [emmett-framework/granian#355] [General]
Your work made me more positive about using django for a new project.

I am not going to use granian immediately, but it feels very good to know that you got my back if my project scales. I have mentioned this project to all of my friends working on huge django codebases. None of them knew about it. When I mentioned it they were extremely enthusiastic.

You probably know it, but for everyone reading this discussion: deploying and monitoring Django is a pain in the butt. And the gunicorn and daphne web servers are not stable in terms of resources consumed. Which is expected, since they have a garbage collector. They keep crashing and have memory leaks.

This leads to a lot of defensive programming and defensive devops. And obviously the elephant in the room: it's very expensive.

I am not saying this project can solve everything, but it's a very good start. I hope this will push people to do more low-level language/python libraries, like we've seen with numpy for example.

My expectations for this project is that it's boring and it just works, like gunicorn (without the bad parts of gunicorn).

Thanks again.

**Comment by gi0baro** [comment]
Thank you for your words!

I think there's still quite a lot of room for improvements, but in order to properly do those having some metrics from the community using Granian on mid/large sized projects would be quite useful.
AFAIK, Mozilla is using Granian for their Django based website (https://github.com/mozilla/bedrock), but I have not received (yet?) any particular feedbacks/metrics.

Please, whenever you have feedbacks or metrics from using Granian in production, share them with the community so we can plan future actions and improvements :)

**Reply by waynehamadi** [reply]
Ok I will let you know
